hibernate-core/src/main/java/org/hibernate/cache/internal/RegionFactoryInitiator.java

```diff
@@ -77,6 +76,43 @@ public RegionFactory initiateService(Map configurationValues, ServiceRegistryImp
 		return regionFactory;
 	}
 
+	@SuppressWarnings("unchecked")
+	private RegionFactory resolveRegionFactory(
+			StrategySelector strategySelector,
+			Object strategyReference,
+			Properties properties) {
+
+		if ( strategyReference == null || RegionFactory.class.isInstance( strategyReference ) ) {
+			// nothing to create; just let strategySelector resolve the RegionFactory
+			return strategySelector.resolveDefaultableStrategy(
+					RegionFactory.class,
+					strategyReference,
+					NoCachingRegionFactory.INSTANCE
+			);
+		}
+
+		final Class<? extends RegionFactory> implementationClass;
+		if ( Class.class.isInstance( strategyReference ) ) {
+			implementationClass = (Class<RegionFactory>) strategyReference;
+		}
+		else {
+			implementationClass = strategySelector.selectStrategyImplementor(
+					RegionFactory.class,
+					strategyReference.toString()
+			);
+		}
+
+		try {
```

Comment:
StrategyCreatorRegionFactoryImpl#create is called directly, instead of via StrategySelector.